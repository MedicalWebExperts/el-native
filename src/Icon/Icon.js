import React from 'react';
import { string, number } from 'prop-types';
import VectorIcon from 'react-native-vector-icons/Ionicons';

const propTypes = {
  /**
   * Color of the icon. Default: Theme white color.
   */
  color: string,
  /**
   * Size of the icon. Default: 16.
   */
  size: number,
  /**
   * What icon to show, see Icon Explorer app or one of the links above. Default: heart.
   */
  name: string,
};

const defaultProps = {
  color: 'white',
  size: 16,
  name: 'ios-heart',
};

const Icon = props => <VectorIcon {...props} />;

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default Icon;
