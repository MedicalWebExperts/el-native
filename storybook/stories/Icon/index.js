import React from 'react';
import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import { Icon } from '../../../src';

storiesOf('Icon', module)
  .addDecorator(getStory => (
    <CenterView style={{ backgroundColor: '#AFAFAF' }}>{getStory()}</CenterView>
  ))
  .add('Default', () => <Icon />)
  .add('Custom color', () => <Icon size={200} color="aquamarine" />)
  .add('Custom size', () => <Icon size={150} />)
  .add('Custom icon', () => <Icon size={200} name="ios-basketball" />)
  .add('Custom all', () => <Icon name="ios-football" size={250} color="azure" />);
