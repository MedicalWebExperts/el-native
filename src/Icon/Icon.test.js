import React from 'react';
import { Platform } from 'react-native';
import renderer from 'react-test-renderer';

import Icon from './Icon';

const customProps = {
  color: 'aquamarine',
  size: 14,
  name: `${Platform.OS === 'ios' ? 'ios' : 'md'}-beer`,
};

describe('Icon Snapshots', () => {
  it('should render a simple Icon', () => {
    const component = renderer.create(<Icon />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a custom Icon', () => {
    const component = renderer.create(<Icon {...customProps} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
