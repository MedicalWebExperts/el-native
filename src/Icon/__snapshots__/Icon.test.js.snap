// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Snapshots should render a custom Icon 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "aquamarine",
        "fontSize": 14,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`Icon Snapshots should render a simple Icon 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#FAFAFA",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;
